cache.go

```diff
@@ -175,11 +180,38 @@ func (c *Cache) Stop() {
 // Gets the number of items removed from the cache due to memory pressure since
 // the last time GetDropped was called
 func (c *Cache) GetDropped() int {
+	return doGetDropped(c.control)
+}
+
+func doGetDropped(controlCh chan<- interface{}) int {
```

Comment:
I added this and `doSyncUpdates` to avoid repetition in the `LayeredCache` implementation. It might be a good idea to take that a little further and have both `Cache` and `LayeredCache` share an internal struct that manages the `promotable`, `deletable`, and `control` channels and then these methods could be implemented by that struct.

Reply:
Worth looking into reduplicating this logic after we're done with this initial re-work. (I generally don't mind a bit of repetition, mind)